spec/pragma.dd

```diff
@@ -76,33 +76,65 @@ $(UL
 
 $(H3 $(LNAME2 crtctor, $(D pragma crt_constructor)))
 
-    $(P This pragma must directly precede an $(D extern(C)) function declaration
-        that must take no argument, even default ones.
-        The function this pragma applies to will be inserted in `.init_array`
-        or `.ctors`, depending on the target and compiler implementation.
-        It is equivalent to GCC's $(LINK2 https://gcc.gnu.org/onlinedocs/gcc-4.7.0/gcc/Function-Attributes.html, `__attribute__((constructor))`).
+    $(P Annotates a function so it is run after the C runtime library is initialized
+        and before the D runtime library is initialized.
     )
------------------
-__gshared int initCount;
 
-pragma(crt_constructor)
-extern(C) void initializer() { initCount += 1; }
------------------
+    $(P The function must:)
+
+    $(OL
+        $(LI be `extern (C)`)
+        $(LI not have any parameters)
+        $(LI not be a non-static member function)
+        $(LI be a function definition, not a declaration (i.e. it must have a function body))
+        $(LI not return a type that has a destructor)
+        $(LI not be a nested function)
```

Comment:
Even `static` ?

Reply:
That's right, even static. Of course this can change, but that's the way it works now.